File: sqfile/src/main/java/com/effective/android/sqfile/scope/app/AppDataFilesScope.java
package com.effective.android.sqfile.scope.app;

import com.effective.android.sqfile.SQFile;
import com.effective.android.sqfile.scope.FileScope;

import java.io.File;

/**
 * /data/data/{packageName}/files 空间
 * created by yummylau on 2020/10/10
 */
public class AppDataFilesScope extends AppFileScope {

    @Override
    public boolean needPermission() {
        return false;
    }

    @Override
    public String path() {
        if (file == null) {
            file();
        }
        return file.getAbsolutePath();
    }

    @Override
    public File file() {
        if (file == null) {
            file = SQFile.getContext().getFilesDir();
        }
        return file;
    }

}

File: sqfile/src/main/java/com/effective/android/sqfile/scope/app/AppDataFilesScope.kt
package com.effective.android.sqfile.scope.app

import android.os.Environment
import android.text.TextUtils
import com.effective.android.sqfile.SQFile
import java.io.File

/**
 * /data/data/{packageName}/files 空间
 * created by yummylau on 2020/10/10
 */
class AppDataFilesScope(splicingPath: String = "") : AppFileScope(splicingPath) {

    override fun root(): File {
        return SQFile.context.filesDir
    }
}